src/components/code/QuestionToggleModal.tsx

```diff
@@ -0,0 +1,190 @@
+import { boolean } from "zod";
+import ModalLayout from "../layout/ModalLayout";
+import { useEffect, useState } from "react";
+import { StyledInput } from "../StyledInput";
+import { Difficulty } from "~/types/global";
+import { StyledButton } from "../StyledButton";
+import { api } from "~/utils/api";
+import useDebounce from "~/hooks/useDebounce";
+
+type ReducedQuestion = {
+  title: string;
+  id: string;
+  category: string;
+  difficulty: Difficulty;
+};
+
+type QuestionToggleModalProps = {
+  questionTitleList: ReducedQuestion[];
+  setQuestionId: (questionTitle: string) => void;
+};
+
+type SelectPageDivProps = {
+  setPage: (inputFunc: (val: number) => number) => void;
+  currentPage: number;
+  totalPages: number;
+  displayedRange: number;
+};
+const SelectPageDiv = ({
+  currentPage,
+  totalPages,
+  displayedRange,
+  setPage,
+}: SelectPageDivProps) => {
+  const [allPages, setAllPages] = useState<number[]>([]);
+  const offset = Math.floor(displayedRange / 2);
+  const actualRange = Math.ceil(displayedRange);
+  useEffect(() => {
+    // update the pages range
+    const newPages = [];
+    const startPage = currentPage - offset < 0 ? 0 : currentPage - offset;

```

Comment:
optional (might be neater)
```suggestion
    const startPage = Math.max(currentPage - offset, 0);
```